archspec/cpu/detect.py

```diff
@@ -323,13 +329,20 @@ def compatibility_check_for_aarch64(info, target):
     vendor = info.get("CPU implementer", "generic")
 
     arch_root = TARGETS[basename]
-    return (
-        (target == arch_root or arch_root in target.ancestors)
+    arch_root_and_vendor = (
+        arch_root == target.family
         and target.vendor in (vendor, "generic")
-        # On macOS it seems impossible to get all the CPU features with syctl info
-        and (target.features.issubset(features) or platform.system() == "Darwin")
     )
 
+    # On macOS it seems impossible to get all the CPU features
+    # with syctl info, but for ARM we can get the exact model
```

Comment:
I have to check whether this comment is correct or we can extract the list of features from `hw.optional.arm.FEAT*`